pkg/template/controller/templateinstance_controller.go

```diff
@@ -28,16 +21,33 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/apiserver/pkg/authentication/user"
 	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
 	authorizationclient "k8s.io/client-go/kubernetes/typed/authorization/v1"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
-	kapi "k8s.io/kubernetes/pkg/apis/core"
-	kclientsetinternal "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
 	"k8s.io/utils/clock"
+
+	templatev1 "github.com/openshift/api/template/v1"
+	buildv1client "github.com/openshift/client-go/build/clientset/versioned"
+	templatev1clienttyped "github.com/openshift/client-go/template/clientset/versioned/typed/template/v1"
+	templatev1informer "github.com/openshift/client-go/template/informers/externalversions/template/v1"
+	templatelister "github.com/openshift/client-go/template/listers/template/v1"
+	"github.com/openshift/origin/pkg/authorization/util"
+	"github.com/openshift/origin/pkg/client/templateprocessing"
 )
 
-const readinessTimeout = time.Hour
+const (
+	readinessTimeout = time.Hour
+
+	WaitForReadyAnnotation    = "template.alpha.openshift.io/wait-for-ready"
```

Comment:
@deads2k candidates for openshift/api (will send a follow up with bump)

Reply:
> mfojtik 5 hours ago  Member
> @deads2k candidates for openshift/api (will send a follow up with bump)

yes, move them there.

Reply:
@deads2k https://github.com/openshift/api/pull/87 (can replace in follow up or just bump the api in this PR after you merge it)